### DISCUSSION TITLE
Add index parameter to the vanX.list function

### DISCUSSION THREAD

**Discussion by ali-alnasser570** [vanjs-org/van#210] [Ideas]
The `vanX.list` function currently takes two parameters for the item and deleter callback:

```
vanX.list(
    div,
    reactiveList,
    (item, deleter) => {}
);

```

I suggest adding an additional parameter for the index to the callback function:

```
vanX.list(
    div,
    reactiveList,
    (item, deleter, index) => {}
);
```

Adding the index parameter would provide developers with valuable information about the position of the item in the list, which can be useful for various scenarios such as custom rendering, animations, or interaction tracking.

**Comment by Tao-VanJS** [comment]
Sounds like a good idea!

Will consider it in the new version of VanX.

**Comment by Tao-VanJS** [comment]
Hi @ali-alnasser570,

This functionality was added into VanX 0.2.0. You can refer to the new API reference (https://vanjs.org/x#api-list) for more information.

**Reply by ali-alnasser570** [reply]
Thank you so much! I was in need for this useful feature
